src/group_theory/perm/cycle/basic.lean

```diff
@@ -929,9 +929,8 @@ end
 
 lemma cycle_factors_finset_eq_finset {σ : perm α} {s : finset (perm α)} :
   σ.cycle_factors_finset = s ↔ (∀ f : perm α, f ∈ s → f.is_cycle) ∧
-    (∃ h : (∀ (a ∈ s) (b ∈ s), a ≠ b → disjoint a b), s.noncomm_prod id
-      (λ a ha b hb, (em (a = b)).by_cases (λ h, h ▸ commute.refl a)
-        (set.pairwise.mono' (λ _ _, disjoint.commute) h ha hb)) = σ) :=
+    ∃ h : (s : set (perm α)).pairwise_disjoint id, s.noncomm_prod id
+      (h.mono' $ λ _ _, disjoint.commute) = σ :=
```

Comment:
I have a fix at [my fork](https://github.com/leanprover-community/mathlib/compare/noncomm_prod_pairwise...alreadydone:mathlib:noncomm_prod_pairwise?expand=1). Should I push here?

Reply:
Sorry! Already had a fix in local.